src/mysql_conn.erl

```diff
@@ -58,7 +58,7 @@
                 affected_rows = 0, status = 0, warning_count = 0, insert_id = 0,
                 transaction_levels = [], ping_ref = undefined,
                 stmts = dict:new(), query_cache = empty, cap_found_rows = false,
-                float_as_decimal = false}).
+                float_as_decimal = false, decode_options = #decode_options{}}).
```

Comment:
I sort of figured this would be a contentious addition, but you never know 'til you try.

I understand your lack of enthusiasm for this :) I also went back and forth on this a few times myself.

I can see a number of potential tunable options come out of this. For example:

* Returning `undefined` instead of `null`
* For date and time fields, skip the conversion to date-time tuples and just returning the string/binary value?
* Options to add custom field-type parsers (like parsing sets, json, etc with 

That said, perhaps just including the option(s) into the `#col` record as suggested minimizes a lot of the headache with adding more options.

At the very least, I could see myself doing the first 2 options (`undefined` and the date stuff).

I don't want to bog down this particular PR (since this particular change is more important to me than the other suggestions), so I'll convert it back down into just the `decode_decimal` as discussed, and re-introduce the `decode_options()` as a future PR.